src/basic/fd-util.h

```diff
@@ -69,6 +65,12 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(DIR*, closedir);
 int fd_nonblock(int fd, bool nonblock);
 int fd_cloexec(int fd, bool cloexec);
 
+static inline void stdio_unset_cloexec(void) {
```

Comment:
just make this a normal function please. This is in no way performance sensitive, it will never appear in inner loops. We have no clear rules when to use inline functions, but I am pretty sure that if something is highly unlikely to be ever used in an inner loop, and where things involve function calls anyway, then they might as well be a proper function.